scripts/populate_ingest/services/ingest.py

```diff
@@ -71,3 +71,9 @@ def project_exists(self, uuid: str) -> bool:
         results = response.json()
         count = results.get('page', {}).get('totalElements', 0)
         return count > 0
+
+    def get_all_projects(self):
+        # get ingest data as json
+        url = f'{self.url}/projects/?page=0&size=1000'
```

Comment:
I see 2 issues here:
- We are using an unprotected endpoint (Which shouldn't be unprotected), so when we finally fix this issue the script will be unusable
- No pagination is added, so when we surpass the 1000 projects this will stop showing new projects

On the pagination issue, I think we could solve it by adding a small function:

```
def yield_all_pages(self, response, size):
    response_json = response.json()
    num_pages = response_json.get('page', {}).get('totalPages', 0)
    url = f"{self.url}/projects/?size={size}"
    for i in range(num_pages):
        response = self.get(f"{url}?page={i}&size={size}")
        yield response.json()
```

Then the next lines would become:
```suggestion
        size = 200
        url = f'{self.url}/projects/'
        response = self.get(url)
        projects = []
        for page in yield_all_pages(response, size):
            projects.extend(page.get('_embedded', {}).get('projects', [])
        return projects
```
I suggested a size but we could keep it as 1000. Anyways, there's still the issue of the authentication

Reply:
Seeing @aaclan-ebi's comment above, disregard this pls :)